ci/Jenkinsfile.android

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env groovy
+library 'status-jenkins-lib@v1.9.24'
+
+/* Options section can't access functions in objects. */
+def isPRBuild = utils.isPRBuild()
+def isNightlyBuild = utils.isNightlyBuild()
+
+pipeline {
+  agent {
+    /* Image with Ubuntu 22.04, QT 6.9.0, Android SDK/NDK, Go, and Nim */
+    docker {
+      label 'linux'
+      image 'statusteam/nim-status-client-build:1.0.0-qt6.9.0-android-ci'
+      alwaysPull true
+    }
+  }
+
+  parameters {
+    booleanParam(
+      name: 'RELEASE',
+      description: 'Decides whether release credentials are used.',
+      defaultValue: params.RELEASE ?: false
+    )
+  }
+
+  options {
+    timestamps()
+    /* Prevent Jenkins jobs from running forever */
+    timeout(time: 45, unit: 'MINUTES')
+    /* manage how many builds we keep */
+    buildDiscarder(logRotator(
+      numToKeepStr: '10',
+      daysToKeepStr: '30',
+      artifactNumToKeepStr: '3',
+    ))
+    /* Allows combined build to copy */
+    copyArtifactPermission('/status-desktop/*')
+    /* Abort old PR builds. */
+    disableConcurrentBuilds(abortPrevious: isPRBuild)
+    disableRestartFromStage()
+  }
+
+  environment {
+    USE_SYSTEM_NIM = "1"
+    GOCACHE = "/tmp/go-cache"
+    GOMODCACHE = "/tmp/go-mod-cache"
+    GOTMPDIR = "/tmp"
+    STATUS_APK_ARTIFACT = "pkg/Status-tablet-${BUILD_NUMBER}.apk"
+    PLATFORM = "android/arm64"
+    QT_ANDROID_PATH = "/opt/qt/6.9.0/android_arm64_v8a"
+    STATUS_APK = "${WORKSPACE}/mobile/bin/android/qt6/Status-tablet.apk"
+  }
+
+  stages {
+    stage('Prepare') {
+      steps {
+        sh 'git submodule update --init --recursive'
```

Comment:
Shouldn't the job itself checkout submodules?

Reply:
I do have recursive checkout enabled in submodule behaviours in Jenkins but this step seems like its needed.